aws-android-sdk-appsync/src/main/java/com/amazonaws/mobileconnectors/appsync/AWSAppSyncClient.java

```diff
@@ -553,8 +551,8 @@ public AWSAppSyncClient build() {
                         throw new RuntimeException("AppSync configuration is missing from awsconfiguration.json");
                     }
 
-                    mServerUrl = appSyncJsonObject.getString("ApiUrl");
-                    mRegion = Regions.fromName(appSyncJsonObject.getString("Region"));
+                    mServerUrl = mServerUrl != null ? mServerUrl : appSyncJsonObject.getString("ApiUrl");
+                    mRegion = mRegion != null ? mRegion : Regions.fromName(appSyncJsonObject.getString("Region"));
```

Comment:
Programmatically set Server URLs and Regions would _always_ be overwritten by the awsconfiguration object